src/metrics/metrics.ts

```diff
@@ -326,49 +342,107 @@ function checkOpening() {}
+
+  for (let moveInfo of moveGenerator) {
+    const { move, board } = moveInfo;
+
+    lastMove = move;
+
+    // Add the current move to the start of the array
+    lastFewMoves.unshift(move);
+
+    // If we have more than 5 moves in the array, remove the oldest one
+    if (lastFewMoves.length > 5) {
+      lastFewMoves.pop();
     }
+
   }
+  console.log("last move: ", lastMove)
+
+  if (lastMove) {
+    lastPieceMoved = lastMove.unambiguousSymbol;
+
+    if (lastMove.originalString.includes('#')) {
+      matingPiece = lastMove.piece;
+      unambigMatingPiece = lastMove.unambiguousSymbol;
 
-  // This is where we DO need to disambiguate, the same piece type but a different piece could provide the assist.
-  // Pausing further work till we've fixed that
+      // Determine the color of the mated king
+      const matedKingColor = lastMove.color === 'w' ? 'b' : 'w';
+      unambigMatedPiece = matedKingColor === 'w' ? 'K' : 'k';
 
+      // If mate see if also assist
+      if (lastFewMoves[2] && lastFewMoves[2].originalString.includes('+') && lastFewMoves[2].unambiguousSymbol !== unambigMatingPiece) {
+        assistingPiece = lastFewMoves[2].piece;
+        unambigAssistingPiece = lastFewMoves[2].unambiguousSymbol;
+
+        // If assist check for hockey assist
+        if (lastFewMoves[4] && lastFewMoves[4].originalString.includes('+') && lastFewMoves[4].unambiguousSymbol !== unambigAssistingPiece && lastFewMoves[4].unambiguousSymbol !== unambigMatingPiece) {
+          hockeyAssist = lastFewMoves[4].piece;
+          unambigHockeyAssistPiece = lastFewMoves[4].unambiguousSymbol;
+        }
+      }
+    } else {
+      // Handle non-checkmate endings here
+      console.log('The game did not end in a checkmate. The last piece that moved was:', lastPieceMoved);
+    }
+  }
+  console.log("mating piece: ", matingPiece);
+  console.log("assisting piece: ", assistingPiece);
+  console.log("hockey assisting piece: ", hockeyAssist);
+  console.log("unambig mating piece: ", unambigMatingPiece);
+  console.log("unambig mated piece: ", unambigMatedPiece);
+  console.log("unambig assisting piece: ", unambigAssistingPiece);
+  console.log("unambig hockey assisting piece: ", unambigHockeyAssistPiece);
+  console.log("last piece moved: ", lastPieceMoved);
+  
   return {
     matingPiece,
     assistingPiece,
     hockeyAssist,
+    unambigMatingPiece,
+    unambigMatedPiece,
+    unambigAssistingPiece,
+    unambigHockeyAssistPiece,
+    lastPieceMoved
   };
 }
 
+// convert PGN string to GameHistoryObject
+export function pgnToGameHistory(pgn: string): GameHistoryMove[] {
+  const chess = new Chess();
+  chess.loadPgn(pgn);
+  return chess.history({ verbose: true });
+}
+
+// convert gameHistory object to a PGN string
+export function gameHistoryToPgn(gameHistory: GameHistoryMove[]): string {
+  const chess = new Chess();
+
+  for (const move of gameHistory) {
+    chess.move(move);
+  }
+
+  return chess.pgn();
+}
+
```

Comment:
These are helpers for the tests right? Can we put them there?

Reply:
yes. i think first used them in metrics.ts for a different function but i dont think they are used there anymore. but any reason to move them over?